src/views/Profile.vue

```diff
@@ -61,28 +100,91 @@ export default {
   name: 'Profile',
   data:() => ({
     user: {
-      name: 'Cesar Villalobos',
-      uid: null,
-      isAdmin: true
     },
+    error: false,
+    searchQuery: '',
+    userDetails: {},
     userUrls: [],
   }),
   methods: {
-    getUserUrls() {
-      const path = "http://127.0.0.1:5000/api/user/1/url"
+    clearResults() {
+      this.userUrls = []
+      this.error = false
+    },
+    copyUrl(signedUrl) {
+      const input = document.createElement('textarea')
+
+      if (this.userUrls.length <= 1) {
+        input.value = `Below is a signed URL good for 7 days that you can use to retrieve your files: \n\n${signedUrl}`
+      } else {
+        input.value = signedUrl
+      }
+      document.body.appendChild(input)
+      input.select()
+      document.execCommand('copy')
+      document.body.removeChild(input)
+
+      console.log(`Copied ${signedUrl} to clipboard`)
+    },
+    fetchSingleUrl() {
+      this.error = false
+      let query = this.searchQuery
+
+      query = query.substring(
+                query.indexOf('datahub/') + "datahub/".length, 
+                query.indexOf(".tar.gz"))
+      const path = `http://127.0.0.1:5000/api/user/${query}/url`
+
+      axios.get(path)
+      .then((res) => {
+        if (!res.data.length) {
+          this.error = true
+        } else {
+          this.userUrls = res.data;
+        }
+      }).catch((err) => {
+        this.error = true
+        console.log(err)
+      })   
+    },
+    fetchAllUrls(){
+      const path = `http://127.0.0.1:5000/api/user/${this.user.uid}/urls`
       axios.get(path)
       .then((res) => {
         this.userUrls = res.data;
       }).catch((err) => {
+        this.error = true
         console.log(err)
       })   
     },
-    clearResults() {
-      this.userUrls = []
+    getAccountDetails() {
+      const path = "http://127.0.0.1:5000/api/account"
+      axios.get(path).then((res) => {
+        this.user = res.data;
+      }).catch((err) => {
+        console.log(err)
+      })
     },
-    copyUrl() {
-      // TO DO: Copy Logic
+    getDayDifference(date) {
+      const today = new Date()
+      const expDate = new Date(date)
+      const diffInMilliseconds = Math.abs(today - expDate)
+
+      return Math.ceil(diffInMilliseconds / (1000 * 60 * 60 *24))
     }
   },
+  beforeMount() {
+    this.getAccountDetails()
+  },
 }
 </script>
+
+<style>
+td {
+  white-space: nowrap; 
+  text-overflow:ellipsis; 
+  overflow: hidden; 
+  max-width:275px;
+}
```

Comment:
@cesarvh - Fyi, the CSS style above is not "scoped" (see Vue docs) and thus will be applied to ALL `<td>` elements in Hartsfield.